cajparser.py

```diff
@@ -397,8 +411,14 @@ def _convert_hn(self, dest):
                     if (image_type_enum == 1):
                         # non-inverted JPEG Images
                         height = -height
+
+                    from PIL import Image as pilimage
+                    with open(".tmp.jpg", "wb") as f:
+                        f.write(image_data)
+                    pim = pilimage.open(".tmp.jpg")
+
                     image_item = (
-                        Colorspace.RGB,
+                        Colorspace[pim.mode],
                         (300, 300),
```

Comment:
Likewise, it should conditional on findings from jpeg header, rather than from a pim.mode array.